### PR TITLE
Toasts

### PR DESCRIPTION
**Pull request by jivey**

### CODE REVIEW

**Comment by jivey** [comment]
Only the basics in here, later on we probably want some dismiss/close buttons, and should probably add a zindex override, etc.

File: src/components/Toast.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    setTimeout(() => {
+      setShow(false);
+      if (onDismiss) {
+        onDismiss(id);
+      }
+    }, dismissAfterMs);
+
+  }, []);
```
[the remark this answers is not in the document]
Comment (jivey) [reply src/components/Toast.tsx:140]: For some reason I missed adding that despite seeing it in the console last week, good catch.

File: src/components/Toast.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  if (!show) { return null; }
+
+  if (dismissAfterMs && dismissAfterMs < DISMISS_AFTER_MS_FLOOR) {
+    dismissAfterMs = DISMISS_AFTER_MS_FLOOR;
+  }
+
+  return <StyledToast
+    dismissAfterMs={dismissAfterMs} inline={inline}>
```
[the remark this answers is not in the document]
Comment (jivey) [reply src/components/Toast.tsx]: Good suggestion, cleans up the logic 👍  (ended up with Math.max)

[the remark this answers is not in the document]
Comment (TomWoodward) [reply src/components/Toast.tsx]: yea because its a floor, oops

File: src/components/Toast.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    const timeoutId = setTimeout(() => {
+      setShow(false);
+      if (onDismiss) {
+        onDismiss(id);
+      }
+    }, dismissAfterMs);
+
+    return () => {
+      clearTimeout(timeoutId);
+      setShow(false);
```
Comment (TomWoodward) [inline src/components/Toast.tsx:145]: i don't think you want to do a state update in here, just clear the timeout

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: src/components/Toast.tsx (hunk 3 above)
Comment (jivey) [reply src/components/Toast.tsx]: I added this because react was complaining about it, and of course now I can't recreate the error 😆

**Review by TomWoodward — APPROVED** [review]

**Pull request merged**